File: src/main/kotlin/net/zhuruoling/omms/controller/mirai/network/instruction/Instruction.java
package net.zhuruoling.omms.controller.mirai.network.instruction;


import com.google.gson.GsonBuilder;

public class Instruction {
    ControllerTypes controllerType;
    String targetControllerName;
    String commandString;

    public boolean matches(ControllerTypes types, String targetControllerName){
        return this.controllerType.equals(types) && this.targetControllerName.equals(targetControllerName);
    }

    public Instruction(){
        controllerType = ControllerTypes.MCDR;
        targetControllerName = "";
        commandString = "";
    }

    public Instruction(ControllerTypes controllerType, String targetControllerName, String commandString) {
        this.controllerType = controllerType;
        this.targetControllerName = targetControllerName;
        this.commandString = commandString;
    }

    public static String  asJsonString(Instruction instruction){
        return new GsonBuilder().serializeNulls().create().toJson(instruction, Instruction.class);
    }

    public static Instruction fromJsonString(String json){
        return new GsonBuilder().serializeNulls().create().fromJson(json, Instruction.class);
    }

    public ControllerTypes getControllerType() {
        return controllerType;
    }

    public void setControllerType(ControllerTypes controllerType) {
        this.controllerType = controllerType;
    }

    public String getTargetControllerName() {
        return targetControllerName;
    }

    public void setTargetControllerName(String targetControllerName) {
        this.targetControllerName = targetControllerName;
    }

    public String getCommandString() {
        return commandString;
    }

    public void setCommandString(String commandString) {
        this.commandString = commandString;
    }

}

File: src/main/kotlin/net/zhuruoling/omms/controller/mirai/network/instruction/Instruction.kt
package net.zhuruoling.omms.controller.mirai.network.instruction

import com.google.gson.GsonBuilder

class Instruction {
    var controllerType: ControllerTypes
    var targetControllerName: String
    var commandString: String
    fun matches(types: ControllerTypes, targetControllerName: String): Boolean {
        return controllerType == types && this.targetControllerName == targetControllerName
    }

    constructor() {
        controllerType = ControllerTypes.MCDR
        targetControllerName = ""
        commandString = ""
    }

    constructor(controllerType: ControllerTypes, targetControllerName: String, commandString: String) {
        this.controllerType = controllerType
        this.targetControllerName = targetControllerName
        this.commandString = commandString
    }

    companion object {
        fun asJsonString(instruction: Instruction?): String {
            return GsonBuilder().serializeNulls().create().toJson(instruction, Instruction::class.java)
        }

        fun fromJsonString(json: String?): Instruction {
            return GsonBuilder().serializeNulls().create().fromJson(json, Instruction::class.java)
        }
    }
}